### ISSUE TITLE
No puedo correr los tests de un proyecto dos veces seguidas porque desaparecen las opciones

### ISSUE THREAD

**Issue by npasserini** [uqbar-project/wollok#1266]
![las opciones desaparecen](https://user-images.githubusercontent.com/4633913/30570602-92e1dc0c-9cb8-11e7-8954-422ecae6544f.gif)

**Comment by npasserini** [comment]
Luego de correr los tests, las opciones en "Run as" desaparecieron.
Si me muevo de tab y vuelvo, aparecen de nuevo.

**Comment by lspigariol** [comment]
lo mismo pasa cuando se corre un wlk (pero no un program), desde el "run as"

(no lo habia visto nunca, porque siempre prefiero darle directo desde el
boton principal)


2017-09-18 21:35 GMT-03:00 Nico Passerini <notifications@github.com>:

**Comment by fdodino** [comment]
Ja, ahí le encontré la vuelta. Si ponés los tests en la misma solapa que el package explorer anda siempre:

![anim](https://user-images.githubusercontent.com/4549002/30609458-d5fb71da-9d51-11e7-8ba6-3af78a6837d9.gif)

Voy a ver qué corno es, pero tiene que ser algo de la configuración del menú, muy malo...

**Comment by fdodino** [comment]
Uf, está difícil.
Fíjense @npasserini  y @lspigariol  que cuando no les aparece las opciones, si en lugar de darle directamente botón derecho, hacen primero un click, el menú se refresca.
¡Malísimo!
Funciona cuando lo tenés en dos solapas porque al volver a la solapa anterior se activa la vista y se refresca el menú.
Voy a investigar qué puede ser...

**Comment by lspigariol** [comment]
No lo probé desde el click izderecho, sino desde el menu desplegable del boton de la barra de herramientas.

**Comment by fdodino** [comment]
Ah, pero ese botón está en el producto cuando se hace el build.
Igual creo que encontré que el problema está acá:

```xtend
def static activate() {
		RunInUI.runInUI [
			    val view = PlatformUI.workbench.activeWorkbenchWindow.activePage.showView(WollokTestResultView.NAME)
			(view as WollokTestResultView).cleanView
			]
	}
```

Porque queda activa la ventana de Wollok Test y esa no te permite abrir el menú Run all tests, hasta que no vuelvas a activar la ventana anterior.

**Comment by npasserini** [comment]
Pero es un problema de secuencia o de cuál es la ventana seleccionada?

2017-09-21 8:57 GMT-03:00 Fernando Dodino <notifications@github.com>:

**Comment by fdodino** [comment]
Cuando el TestResultView se activa se  pierde el contexto de la ventana anterior, cuando hacés botón derecho no se activa el menú contextual de nuevo, sino hasta que hacés un click izquierdo: con eso activás de nuevo el project explorer y el botón derecho funciona. A mí siempre me anda porque yo agrupo en un mismo conjunto de solapas al Project Explorer y a los Wollok Test. Entonces eso hace que para volver a seleccionar tengo que activar de nuevo el Project Explorer. Ya lo arreglé para tu configuración, falta que funcione bien para la mía.

**Comment by npasserini** [comment]
En realidad yo también siempre lo uso como lo tenés vos, lo moví el otro
día porque etsaba teniendo un problema con el monitor.
Que yo sepa, la configuración que yo estaba usando en ese momento no es
habitual, ergo, si sólo pasa en ese caso, podríamos decidir que esto tiene
bastante baja prioridad.

2017-09-21 14:05 GMT-03:00 Fernando Dodino <notifications@github.com>:

**Comment by fdodino** [comment]
Voy a probar un poco más, y si no sale lo voy a dejar para más adelante porquie lo que logré fue detectar al activarse la ventana de Wollok Tests cuál era la ventana activa anterior. Cuando corren los tests vuelvo a activar esa ventana activa. El tema es que si lo ponés en la misma solapa, el efecto que tiene es que no ves los tests, es peor eso que directamente recordar hacer un click izquierdo y luego el derecho.

El problema es que 

* Eclipse no me deja ver si una vista está "tapada" dentro de una solapa, 
* ni siquiera puedo saber si están compartiendo la misma solapa
* tampoco encontré un lostFocus / blur (en algunas páginas menciona un listener que no me funcionó ni a palos en la versión que tenemos)

Esto complica para saber qué tipo de configuración tiene y actuar en consecuencia.
Cualquier novedad les chiflo pero está complicado.
